parse.go

```diff
@@ -11,311 +11,301 @@ import (
 	"strings"
 )
 
-// ConfigFileParser interprets the config file represented by the reader
-// and calls the set function for each parsed flag pair.
-type ConfigFileParser func(r io.Reader, set func(name, value string) error) error
-
-// Parse the flags in the flag set from the provided (presumably commandline)
-// args. Additional options may be provided to have Parse also read from a
-// config file, and/or environment variables, in that priority order.
-func Parse(fs *flag.FlagSet, args []string, options ...Option) error {
-	var c Context
-	for _, option := range options {
-		option(&c)
+// Parse the flag set with the provided args. [Option] values can be used to
+// influence parse behavior. For example, options exist to read flags from
+// environment variables, config files, etc.
+//
+// The fs parameter must be of type [Flags] or [*flag.FlagSet]. Any other type
+// will result in an error.
+func Parse(fs any, args []string, options ...Option) error {
```

Comment:
I'm not super happy with `fs any` but I judge that maintaining support for `ff.Parse(<*flag.FlagSet>, ...)` is important enough to justify complecting the API in this way. I could be convinced otherwise, especially if there were a better design that I haven't thought of...

Reply:
I would have loved to use generics to say `fs interface{ ff.FlagSet | *flag.FlagSet }` — but you can't define a union over anything but concrete types. I guess it makes sense why, but it's unfortunate in this case.

Reply:
I agree that seeing `any` in a signature where there is actually a runtime constraint is uncomfortable, but that's at least partially due to the permissive connotation. While we can't (_yet_) be more specific with general interfaces, you might at least be able to sharpen up the perceptual edges with a type alias like:

```
// AnyFlagSet must be one of { [ff.FlagSet] | [*flag.FlagSet] }
type AnyFlagSet = any
```

This at least calls out the restriction for now, even if it doesn't have any compile-time checking.

Reply:
Or, alternatively, `FlagSetAny`, depending on whether you think of it as "an `any` that's a flagset" or "[the] FlagSet types' `any`" :D

Reply:
I like this a lot!

Reply:
#115 for your perusal!